src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -343,86 +376,160 @@ func (r *ClusterReconciler) WithClusterDomain(
 	return r
 }
 
+// nolint:funlen,gocyclo // External nodes list should be refactored
 func (r *ClusterReconciler) createExternalNodesList(
 	ctx context.Context,
 	pods []corev1.Pod,
 	pandaCluster *redpandav1alpha1.Cluster,
 	nodePortName types.NamespacedName,
-) (
-	external, externalAdmin, externalProxy []string,
-	proxyIngress string,
-	err error,
-) {
+) (*redpandav1alpha1.NodesList, error) {
 	externalKafkaListener := pandaCluster.ExternalListener()
 	externalAdminListener := pandaCluster.AdminAPIExternal()
 	externalProxyListener := pandaCluster.PandaproxyAPIExternal()
-	if externalKafkaListener == nil && externalAdminListener == nil {
-		return []string{}, []string{}, []string{}, "", nil
+	schemaRegistryConf := pandaCluster.Spec.Configuration.SchemaRegistry
+	if externalKafkaListener == nil && externalAdminListener == nil && externalProxyListener == nil &&
+		(schemaRegistryConf == nil || schemaRegistryConf.External != nil && !schemaRegistryConf.External.Enabled) {
```

Comment:
this is very complicated if and I am not even sure it's right :) can you please change this to `!pandaCluster.Spec.IsSchemaRegistryExternallyAvailable` I think that should be enough, right?

Reply:
Not exactly. If `schemaRegistryConf` is `nil` then the:
```go
if externalKafkaListener == nil && externalAdminListener == nil && externalProxyListener == nil &&
   !pandaCluster.IsSchemaRegistryExternallyAvailable(){
}
```

Is false. Because of that we try to report external list which is failing.

Reply:
The
```go
r.Spec.Configuration.SchemaRegistry == nil ||
r.Spec.Configuration.SchemaRegistry != nil &&
r.Spec.Configuration.SchemaRegistry.External != nil &&
r.Spec.Configuration.SchemaRegistry.External.Enabled
```

was corrected by goland to
```go
r.Spec.Configuration.SchemaRegistry == nil ||
r.Spec.Configuration.SchemaRegistry.External != nil &&
r.Spec.Configuration.SchemaRegistry.External.Enabled
```